jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/TaskContentServiceImpl.java

```diff
@@ -93,7 +94,10 @@ public long addOutputContent(long taskId, Map<String, Object> params) {
         }
         taskEventSupport.fireBeforeTaskOutputVariablesChanged(task, context, initialContent);
         ((InternalTaskData)task.getTaskData()).setTaskOutputVariables(params);
+                
+        ((InternalTaskData)task.getTaskData()).setTaskOutputVariables(params);        
         taskEventSupport.fireAfterTaskOutputVariablesChanged(task, context, params);
+        persistenceContext.updateTask(task);
```

Comment:
this is not really needed for JPA as it's anyway "recorded" in persistence context but since the integration with other data stores is done through persistence layer it needs to be there for consistency reason.